microprofile/weld/weld-core-impl/src/main/java/org/jboss/weld/bean/proxy/ProxyFactory.java

```diff
@@ -1074,4 +1068,24 @@ private static String getDefaultPackageReason(Class<?> clazz) {
         }
         return null;
     }
+
+    /**
+     * Delegates proxy creation via {@link ProxyServices} to the integrator or to our own implementation.
```

Comment:
Minor note/question: I see this delegating to `ProxyServices` in all cases and never to "our own implementation".